### DISCUSSION TITLE
Google Drive挂载图文教程-2022/9/30

### DISCUSSION THREAD

**Discussion by libsgh** [libsgh/PanIndex#359] [General]
1. 打开PanIndex Tool获取工具，[https://pt.noki.icu/](https://pt.noki.icu/)或workers域名（永久）[https://mgaa.noki.workers.dev/](https://mgaa.noki.workers.dev/)
![image](https://user-images.githubusercontent.com/25656088/193232235-03fada66-9ef0-47c1-9386-a30f61d33497.png)
2. [创建项目](https://console.cloud.google.com/apis/dashboard)
![image](https://user-images.githubusercontent.com/25656088/193232661-0adcd5da-af76-4e1f-b00a-33c789d494c8.png)
![image](https://user-images.githubusercontent.com/25656088/193232887-5f11539b-2a49-4f8e-81cf-0391bbbef5c3.png)
3. 启用API和服务
![image](https://user-images.githubusercontent.com/25656088/193233332-de56330e-c46d-4481-8cf4-f835f229c55c.png)
##### 搜索谷歌网盘API
![image](https://user-images.githubusercontent.com/25656088/193233624-1b0e52aa-ecfe-4172-bc96-b89770f11253.png)
##### 启用API
![image](https://user-images.githubusercontent.com/25656088/193233811-3010ed8a-74e5-41e6-905f-39edd59791a6.png)
4. 配置OAuth屏幕
![image](https://user-images.githubusercontent.com/25656088/193234835-e18e5029-1056-4fe7-9e1f-4b9df6a8ec13.png)
##### 填写应用名称、电子邮件
![image](https://user-images.githubusercontent.com/25656088/193235833-666ad393-9bbb-4649-af6e-ed978a9fd4d3.png)
##### 添加已授权的网域，也就是你PanIndex的域名，这里要填写顶级域名
![image](https://user-images.githubusercontent.com/25656088/193236084-66f2e79b-1a19-45b5-bf18-ee0288d2a1eb.png)
##### 返回摘要信息，我们将应用发布
![image](https://user-images.githubusercontent.com/25656088/193236988-d3959cc4-d79e-4c70-aefe-b67debdc9d22.png)
5. 创建客户端凭据，获取client_id、client_secret、redirect_uri并保存
![image](https://user-images.githubusercontent.com/25656088/193237234-7dd0a258-2221-46aa-9659-66a3e8196edc.png)
![image](https://user-images.githubusercontent.com/25656088/193237364-b81fb2b4-4204-4ac9-971a-32bf2dae0317.png)
##### 应用类型选择WEB应用
![image](https://user-images.githubusercontent.com/25656088/193237566-c6656b20-7b29-403f-976a-bf80cf6b3cd4.png)
##### JavaScript 来源、已获授权的重定向 URI都填写PanIndex Tool工具地址
![image](https://user-images.githubusercontent.com/25656088/193240774-a13d9fe7-13a4-4d47-83e3-5f9722105a9a.png)
##### 保存密钥ID和密钥值，可以下载JSON以备不时之需
![image](https://user-images.githubusercontent.com/25656088/193239221-077a7716-543e-43ca-a5cb-7a35f223333e.png)
6. 返回PanIndex Tool工具，输入上一步获取的客户端密钥ID，**ID和值不是一个东西，请不要输错**
![image](https://user-images.githubusercontent.com/25656088/193239640-47a75b53-35d4-4e56-967c-80f1a935157d.png)
**授权中有一些弹出提示，请确认继续**，授权成功后会返回PanIndex Tool
![image](https://user-images.githubusercontent.com/25656088/193241589-d0a790e2-7c0c-4c2d-86ea-bce3527b45f3.png)
7. 点击确认后，授权CODE会自动填入，输入之前获取的客户端密钥值，**ID和值不是一个东西，请不要输错**
8. 成功后会跳转一个地址返回的信息中的`refresh_token`**至关重要**请妥善保存，因为这个token只在第一次获取的时候才会有，后面只能获取到`access_token`了
> 我这里因为装了JSON格式化显示插件可能与你显示的不太一样，你可以复制信息随便打开一个JSON格式化网站，格式化后获取信息，例如：https://www.bejson.com/

![image](https://user-images.githubusercontent.com/25656088/193242360-8e003b9d-f728-422e-a3f4-58f61b245856.png)
9. 至此，Google Drive申请完毕，我们可以去PanIndex后台挂载了，依次填入上面步骤中获取的信息（如果挂载根目录ID请设置成`root`），**一定要开启流量中转，不然文件下载是需要登录google的，切记**
![image](https://user-images.githubusercontent.com/25656088/193243899-a1030531-1a77-49e1-af89-c7757dacfcff.png)
10. End~

**Comment by yuanying1199** [comment]
提示一下：
看到第9步打码的**根目录ID** ，终于搞清楚我配置咋会报[错误](https://github.com/libsgh/PanIndex/discussions/353)
第9步 **根目录ID** 必须填上，Google云盘的根目录ID是 官网 URL类似这样：https://drive.google.com/drive/u/1/folders/1J0nIHPVPK9ZykNZfAjryUJi1PPA4jHi 的最后字符串 1J0nIHPVPK9ZykNZfAjryUJi1PPA4jHi

**Reply by libsgh** [reply]
> 提示一下： 看到第9步打码的**根目录ID** ，终于搞清楚我配置咋会报[错误](https://github.com/libsgh/PanIndex/discussions/353) 第9步 **根目录ID** 必须填上，Google云盘的根目录ID是 官网 URL类似这样：https://drive.google.com/drive/u/1/folders/1J0nIHPVPK9ZykNZfAjryUJi1PPA4jHi 的最后字符串 1J0nIHPVPK9ZykNZfAjryUJi1PPA4jHi

这里只是讲怎么获取`refresh_token`，网盘目录ID怎么获取文档有说明的，如果你想挂载GD的根目录，请设置目录ID为`root`。
